src/test/java/uk/gov/companieshouse/api/accounts/controller/NoteControllerTest.java

```diff
@@ -0,0 +1,351 @@
+                .body(responseObject.getData());
+        when(apiResponseMapper.mapGetResponse(responseObject.getData(), request))
+                .thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse =
+                noteController.get(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.FOUND, responseEntity.getStatusCode());
+        assertEquals(note, responseEntity.getBody());
+    }
+
+    @Test
+    @DisplayName("Get Note resource - data exception thrown")
+    void getNoteResourceDataException() throws DataException {
+
+        when(request.getAttribute(AttributeName.TRANSACTION.getValue())).thenReturn(transaction);
+        when(accountsNoteConverter.getAccountsNote(accountType, noteType)).thenReturn(accountingNoteTypeWithExplicitValidation);
+
+        DataException dataException = new DataException("");
+        when(noteService.find(accountingNoteTypeWithExplicitValidation, COMPANY_ACCOUNTS_ID)).thenThrow(dataException);
+
+        ResponseEntity responseEntity =
+                ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+
+        when(apiResponseMapper.getErrorResponse()).thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse = noteController.get(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
+        assertNull(responseEntity.getBody());
+    }
+
+    @Test
+    @DisplayName("Delete Note resource - success")
+    void deleteNoteResourceSuccess() throws DataException {
+
+        when(request.getAttribute(anyString())).thenReturn(transaction);
+        when(accountsNoteConverter.getAccountsNote(accountType, noteType)).thenReturn(accountingNoteTypeWithExplicitValidation);
+
+        ResponseObject<Note> responseObject = new ResponseObject<>(ResponseStatus.UPDATED,
+                note);
+
+        when(noteService.delete(accountingNoteTypeWithExplicitValidation, COMPANY_ACCOUNTS_ID, request))
+                .thenReturn(responseObject);
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.NO_CONTENT)
+                .build();
+        when(apiResponseMapper.map(responseObject.getStatus(), responseObject.getData(),
+                responseObject.getErrors()))
+                .thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse =
+                noteController.delete(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.NO_CONTENT, responseEntity.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Delete Note resource - data exception thrown")
+    void deleteNoteResourceDataException() throws DataException {
+
+        when(request.getAttribute(anyString())).thenReturn(transaction);
```

Comment:
Please keep this consistent across methods .
 when(request.getAttribute(AttributeName.TRANSACTION.getValue()))

Reply:
Done